core/eval.go

```diff
@@ -1526,6 +1526,42 @@ func evalCommandGetKeys(args []string) []byte {
 
 	return Encode(keys, false)
 }
+func evalRename(args []string) []byte {
+
+	if len(args) != 2 {
+		return Encode(errors.New("ERR wrong number of arguments for 'RENAME' command"), false)
+	}
+	sourceKey := args[0]
+	destKey := args[1]
+
+	//if Source and Destination Keys are same return RESP encoded ok
+	if sourceKey == destKey {
+		return RESP_OK
+	}
+
+	sourceObj := Get(sourceKey)
+
+	// if Source key does not exist, return RESP encoded nil
+	if sourceObj == nil {
+		return Encode("(error) ERR no such key", false)
+	}
+
+	destObj := Get(destKey)
+
+	//Delete the destination key if exists
+	if destObj != nil {
+		Del(destKey)
+	}
+
+	// Rename the key by putting the source key's value in the destination key
+	Put(destKey, sourceObj)
+
+	//Deleting the source Key
+	Del(sourceKey)
```

Comment:
This operation needs to be atomic, which may not be easily possible in the current locking approach. Please add a TODO here so we can look at it later.
@soumya-codes I was wondering if you have any suggestions to handling such cases?

Reply:
Alternatively, we could introduce a new store-level API which does an atomic rename.

Reply:
I would vote for the second approach, i.e introducing a new store level API.
Considering the first approach, atomicity across different operations could be handled in a transaction. Transactions would hold locks to the keys being changed.

Reply:
@JyotinderSingh, @gauravsarma1992 I agree with the second approach. A store-level API that does an atomic rename seems more appropriate.